Content.Server/Supermatter/Systems/SupermatterSystem.cs

```diff
@@ -0,0 +1,546 @@
+
+            //before we actually start counting down, check to see what delam type we're doing.
+            if (!sMcomponent.FinalCountdown)
+            {
+                //if we're in atmos
+                if (mixture is { })
+                {
+                    //Absorbed gas from surrounding area
+                    var absorbedGas = mixture.Remove(sMcomponent.GasEfficiency * mixture.TotalMoles);
+                    var absorbedTotalMoles = absorbedGas.TotalMoles;
+                    //if the moles on the sm's tile are above MolePenaltyThreshold
+                    if (absorbedTotalMoles >= sMcomponent.MolePenaltyThreshold)
+                    {
+                        sMcomponent.DelamType = DelamType.Singulo;
+                        _chat.TrySendInGameICMessage(uid, Loc.GetString("supermatter-delamination-overmass"),
+                            InGameICChatType.Speak, hideChat: true);
+                    }
+                }
+                else
+                {
+                    sMcomponent.DelamType = DelamType.Explosion;
+                    _chat.TrySendInGameICMessage(uid, Loc.GetString("supermatter-delamination-default"),
+                        InGameICChatType.Speak, hideChat: true);
+                }
+            }
+
+            sMcomponent.FinalCountdown = true;
+
+            sMcomponent.DelamTimerAccumulator += frameTime;
+            sMcomponent.SpeakAccumulator += frameTime;
+            var roundSeconds = sMcomponent.DelamTimerTimer - (int) Math.Floor(sMcomponent.DelamTimerAccumulator);
+
+            //we're more than 5 seconds from delam, only yell every 5 seconds.
+            if (roundSeconds >= sMcomponent.YellDelam && sMcomponent.SpeakAccumulator >= sMcomponent.YellDelam)
+            {
+                sMcomponent.SpeakAccumulator -= sMcomponent.YellDelam;
+                _chat.TrySendInGameICMessage(uid,
+                    Loc.GetString("supermatter-seconds-before-delam", ("Seconds", roundSeconds)),
+                    InGameICChatType.Speak, hideChat: true);
+            }
+            //less than 5 seconds to delam, count every second.
+            else if (roundSeconds < sMcomponent.YellDelam && sMcomponent.SpeakAccumulator >= 1)
+            {
+                sMcomponent.SpeakAccumulator -= 1;
+                _chat.TrySendInGameICMessage(uid,
+                    Loc.GetString("supermatter-seconds-before-delam", ("Seconds", roundSeconds)),
+                    InGameICChatType.Speak, hideChat: true);
+            }
+
+            //TODO: make tesla(?) spawn at SupermatterComponent.PowerPenaltyThreshold and think up other delam types
+            //times up, explode or make a singulo
+            if (!(sMcomponent.DelamTimerAccumulator >= sMcomponent.DelamTimerTimer))
+                return;
+
+            if (sMcomponent.DelamType == DelamType.Singulo)
+            {
+                //spawn a singulo :)
+                EntityManager.SpawnEntity("Singularity", xform.Coordinates);
```

Comment:
Hardcoded entity evil

Just make the Entity a datafield and make the delam enum 'spawn entity' or something.

Delam that spawns a Cluwne.
Or a mcnugged.